sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java

```diff
@@ -96,8 +108,28 @@
               public Void apply(Iterable<Void> iter) {
                 return null; // ignore input
                 }
-            }))
-        .apply(Keys.<T>create());
+            }));
+    if (!acrossPanes) {
+      return combined.apply("SinglePaneDistinct", Keys.<T>create());
+    } else {
+      return combined.apply("StatefulDistinct", ParDo.of(new DoFn<KV<T, Void>, T>() {
+        // Keep track of whether we've seen a given key yet.
+        @StateId("exists")
+        private final StateSpec<CombiningState<Integer, int[], Integer>> stateSpec =
+            StateSpecs.combining(Sum.ofIntegers());
```

Comment:
You would write a `DistinctCombineFn` that basically just does `||` and stores in a `boolean[]` since this _is_ a boolean implemented as an integer. I had thought it would be more readable, and be guaranteed to always store in one byte. But as long as the input coder is `VarIntCoder` it should still be one byte. So I can skip that.